nft-example/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+#![no_std]
+#![feature(const_btree_new)]
+
+use codec::{Decode, Encode};
+use gstd::{debug, exec, msg, prelude::*, ActorId};
+use primitive_types::{H256, U256};
+use scale_info::TypeInfo;
+
+pub mod payloads;
+pub use payloads::{ApproveForAllInput, ApproveInput, InitConfig, TransferInput};
+
+pub mod state;
+pub use state::{State, StateReply};
+
+use non_fungible_token::base::NonFungibleTokenBase;
+use non_fungible_token::{Approve, ApproveForAll, NonFungibleToken, Transfer};
+
+const GAS_RESERVE: u64 = 500_000_000;
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+#[derive(Debug, Decode, TypeInfo)]
+pub enum Action {
+    Mint,
+    Burn(U256),
+    Transfer(TransferInput),
+    Approve(ApproveInput),
+    ApproveForAll(ApproveForAllInput),
+    OwnerOf(U256),
+    BalanceOf(H256),
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum Event {
+    Transfer(Transfer),
+    Approval(Approve),
+    ApprovalForAll(ApproveForAll),
+    OwnerOf(H256),
```

Comment:
Suggestion: May be user `ActorId` instead of `H256`. But can be done in separate PR

Reply:
I will replace it in that PR, thank you